### ISSUE TITLE
RMS: Registration and/or Payment Status

### ISSUE THREAD

**Issue by game7** [game7/sportified#101]
Need to come up with the various states that are required for a registration.  Currently once the user clicks the "register" button on the product detail page we create the new registration and redirect the user to "checkout".  Checkout decides how to route the user based on a) if there are any forms required and whether each has been completed and b) whether payment is required and has been completed.  In this sense there actually is an implicit state for each registration but this state is derived via inspection, which can limit how and when we can use this state.  I think we need to have an explicit state property for each registration that is persisted to the DB and updated accordingly as changes are made to the registration.  What are those states?

Currently the "checkout" controller repeatedly inspects the registration to determine state and routes to the next step that is required to push registration closer to "completed".  The checkout controller is probably the appropriate place to house the step-by-step flow logic but it will be important to move the state back to the registration so that we can orchestrate different flows -- for example, we had talked previously about scenarios such as:

* register online, forms online, pay in person
* invite to register, forms online, pay online
* register online, pay online, subsequently complete forms online

In short, by storing state within the registration we can start/stop/resume from different states in different order.
